### PR TITLE
Fixed GCC warnings & errors.

### PR DESCRIPTION
**Pull request by bkaradzic**

### CODE REVIEW

File: lib/Remotery.c [hunk 1]
```diff
@@ -508,7 +508,7 @@ static void WriteFence()
 
 // Get a shared value with acquire semantics, ensuring the read is complete
 // before the function returns.
-static void* LoadAcquire(void* volatile const* addr)
+void* LoadAcquire(void* volatile const* addr)
```
Comment (dwilliamson) [inline lib/Remotery.c]: Why remove the static?

**Comment by dwilliamson** [comment]
Thanks for the fixes. Left just one comment.

**Comment by bkaradzic** [comment]
It's removed because function is not referenced anywhere and it complains that it's static and unused (can't be referenced outside of compilation unit).

**Comment by dwilliamson** [comment]
OK, just pulled but getting errors for CRTMalloc/Free/Realloc when compiling in C mode. Will fix.

**Comment by bkaradzic** [comment]
Oh sorry, I was just building with C++.

**Pull request merged**